.github/workflows/linux.yml

```diff
@@ -355,25 +355,16 @@ jobs:
           source ${OV_INSTALL_DIR}/setupvars.sh
           cmake --install ./build/ --config ${{ matrix.build-type }} --prefix ${OV_INSTALL_DIR}
 
-      - name: Build samples (Release)
-        if: ${{ 'Release' == matrix.build-type }} # build_samples enforces Release build
+      - name: Build samples
         run: |
           ${OV_INSTALL_DIR}/samples/cpp/build_samples.sh -i ${{ github.workspace }}/s\ pace
 
-      - name: Build samples (Debug)
-        if: ${{ 'Release' != matrix.build-type }}
-        run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${OV_INSTALL_DIR}/samples/cpp/ -B ./samples\ build/ && cmake --build ./samples\ build/ --config ${{ matrix.build-type }} -j
-          cmake --install ./samples\ build/ --config ${{ matrix.build-type }} --component samples_bin --prefix s\ pace
```

Comment:
Other than Release builds must be handled explicitly because 
> build_samples enforces Release build

 See the comment that you deleted :)